### PR TITLE
add holy damage and update weapon procs

### PR DESCRIPTION
**Pull request by wershlak**
Add holy damage and update the demonslayer proc to be useful. I refactored a bit and updated the Deikhan sword proc too.

### CODE REVIEW

**Review by drewhershey — COMMENTED** [review]
Found a few things I wasn't sure about. I didn't actually compile or boot the branch up, figuring you did in-game testing already.

File: code/code/spec/spec_objs_deikhan_sword.cc [hunk 1]
```diff
@@ -151,10 +199,23 @@ int deikhanSword(TBeing* vict, cmdTypeT cmd, const char* arg, TObj* o, TObj*) {
     }
 
     if (!::number(0, 3)) {
+      if (!::number(0, 2) && isEvil(vict))
+        return doSmiteEvil(ch, vict, o);
       return doHarm(ch, vict, o);
```
[the remark this answers is not in the document]
Comment (wershlak) [reply code/code/spec/spec_objs_deikhan_sword.cc:204]: 25% chance to proc and when it procs 33% of those should be doSmiteEvil while 66% are doHarm. And only if the appropriate victim race of course.

Does the code match that intent?

[the remark this answers is not in the document]
Comment (drewhershey) [reply code/code/spec/spec_objs_deikhan_sword.cc:204]: Yep, makes sense, figured it was probably something like that.

File: code/code/obj/obj_low.cc [hunk 2]
```diff
@@ -221,7 +221,7 @@ int ObjectEvaluator::getStatPointsRaw() {
     4.0,  // IMMUNE_NONMAGIC
     0.5,  // IMMUNE_PLUS1
     1.0,  // IMMUNE_PLUS2
-    1.5,  // IMMUNE_PLUS3
+    1.5,  // IMMUNE_HOLY
```
[the remark this answers is not in the document]
Comment (wershlak) [reply code/code/obj/obj_low.cc]: Fixed

**Review by drewhershey — DISMISSED** [review]
Looks good to me

**Review by drewhershey — APPROVED** [review]
Looks even better

**Pull request merged**